flang/lib/Lower/ReductionRuntime.cpp

```diff
@@ -110,3 +166,66 @@ Fortran::lower::genAny(Fortran::lower::FirOpBuilder &builder,
    
   return builder.create<fir::CallOp>(loc, anyFunc, args).getResult(0);
 }
+
+/// Generate call to Maxloc intrinsic runtime routine. This is the version
+/// that does not take a dim argument.
+void
+Fortran::lower::genMaxloc(Fortran::lower::FirOpBuilder &builder, 
+                          mlir::Location loc, 
+                          mlir::Value resultBox,
+                          mlir::Value arrayBox, 
+                          mlir::Value maskBox, 
+                          mlir::Value kind, 
+                          mlir::Value back) {
```

Comment:
Nit: I am not sure but the formatting looks weird to me, did you run clang-format on this too ? As a general practice, we always run clang-format on the .h and .cpp files of pull requests so that the formatting is always consistent (and so that there is no need to care too much about the formatting while coding).

You can pass several files at the same time to clang-format to avoid losing too much time with this `clang-format -i file1.cpp file2.cpp ....`, and I think there is a git plugging that can run it for you when doing git commit, but I am not using it.